### PR TITLE
Updated create-retention-policies.md

### PR DESCRIPTION
**Pull request by YuriySamorodov**
Corrected OneDrive address naming convention description

### CODE REVIEW

**Comment by yogkumgit** [comment]
@JohanFreelancer9 - kindly do the copy editor review for this PR. Thanks

File: microsoft-365/compliance/create-retention-policies.md [hunk 1]
```diff
@@ -204,9 +204,10 @@ Although the retention policy is applied at the site level, only documents have
...
 
-To specify individual OneDrive accounts to include or exclude, the URL has the following format: `https://<tenant name>-my.sharepoint.com/personal/<user_name>_<tenant name>_com`
+To specify individual OneDrive accounts to include or exclude, the URL has the following format: `https://<tenant name>-my.sharepoint.com/personal/<userprincipalname>`
 
-For example, for a user in the contoso tenant that has a user name of "rsimone": `https://contoso-my.sharepoint.com/personal/rsimone_contoso_onmicrosoft_com`
+> [!NOTE] All period characters in <userprincipalname> should be replaced with underescores if any.
```
Comment (JohanFreelancer9) [inline microsoft-365/compliance/create-retention-policies.md]:
```suggestion
> [!NOTE]
> All period characters in <userprincipalname> should be replaced with underscores if any. As an example, the OneDrive address for a user with the UserPrincipalName of "adam.smith@contoso.com" would be `https://contoso-my.sharepoint.com/personal/adam_smith_contoso_com`.
```

File: microsoft-365/compliance/create-retention-policies.md [hunk 2]
```diff
@@ -204,9 +204,10 @@ Although the retention policy is applied at the site level, only documents have
...
+> As an example, OneDrive address for a user with UserPrincipalName of "adam.smith@contoso.com" would be `https://contoso-my.sharepoint.com/personal/adam_smith_contoso_com`
```
Comment (JohanFreelancer9) [inline microsoft-365/compliance/create-retention-policies.md]:
```suggestion
```

**Review by JohanFreelancer9 — CHANGES_REQUESTED** [review]
Copy review - @YuriySamorodov, kindly commit the suggested changes. Many thanks.

**Review by JohanFreelancer9 — APPROVED** [review]

**Comment by yogkumgit** [comment]
@cabailey Please merge this PR. Thanks

**Comment by cabailey** [comment]
This is a technical change & I don't see confirmation from an authoritative source.  @YuriySamorodov, are you a Microsoft employee or can you provide a reference for this change, such as a support ticket?

**Comment by YuriySamorodov** [comment]
Hi @cabailey ,

Thanks for the update. I am not Microsoft employee, however I have a vast experience with Office365 as an administrator and I know by heart examples in the article are inaccurate.
Although I have many tickets opened with Microsoft, none of those are related. However, I believe the following proves my point:

- There is no ```UserName``` attribute neither in AD or AzureAD. Few similar ones like ```Name``` or ```SamAccountName``` but none exactly the same.
- If you have M365 license you may want to check it for yourself. Just open your OneDrive and make a note of the URL.
If your tenant is a production environment, it is more than likely you will see your email address with all special characters replaced with underscores instead of ```onmicrosoft.com``` portion.
Here is mine:
image
It's obvious blurred part in the end is way too short for "onmicrosoft_com"
However here is what I get if my domain is replaced with "<tenantname>_onmicrosoft_com":
image
And just for the sake of proof below is a bunch of OneDrive sites pulled with PnP.PowerShell:
image
As you may see, none of the above users have "onmicrosoft_com" in their OneDrive site addresses. Instead it represents their ```UserPrincipalName``` with underscores instead of special characters.

Hope this helps.

**Comment by cabailey** [comment]
@YuriySamorodov Thank you for your detailed explanation & sharing your experience. I've been able to run this to ground with an authoritative source from the OneDrive team and updated the article accordingly. It seems the "user name" came from the UI field when you create a new user in the Azure portal, and the example did match exactly against my test tenant (still using the default of "onmicrosoft.com"). However, the explanation and syntax representation was incorrect. I also learned that trying to predict the URL isn't always reliable because of potential conflict detection at the time of creation. The updated information:

To specify individual OneDrive accounts to include or exclude, the URL for a user's OneDrive is usually in the following format. For the user principal name (UPN), any special characters such as a period, comma, space, and the at sign ("@") are converted to underscores ("_"): https://<tenant name>-my.sharepoint.com/personal/<user principal name>

For example, for a user in the Contoso tenant who has a UPN of "rsimone@contoso.onmicrosoft.com": https://contoso-my.sharepoint.com/personal/rsimone_contoso_onmicrosoft_com. However, numbers or GUIDs can be appended when conflicts are detected.

It's always best to confirm a user's URL for their OneDrive account, which you can do with the Microsoft 365 admin center, or PowerShell. For more information, see Get a list of all user OneDrive URLs in your organization.

**Comment by YuriySamorodov** [comment]
Thanks @cabailey, much appreciated.
Please note though ```UserPrincipalName``` is one word which represents a specific attribute in AD.

**Comment by cabailey** [comment]
For easier readability, we used complete words - not everybody works at the level of AD attribute names. The term is "user principal name", which is represented by "User Principal Name" in the Azure AD portal, and UserPrincipalName for the Azure AD attribute. I don't think this will trip up anybody.

**Comment by YuriySamorodov** [comment]
Dear @cabailey ,
First and foremost please don't take my comments and PRs personal. It's not my intention to steal your job or devaluate your technical writer skills. The only reason I spend my time suggesting changes is to improve Microsoft documentation.
Believe me or not, the vast majority of users coming to https://docs.microsoft.com are either admins or developers. Especially when it comes to Microsoft 365 **Admin** Center related articles. And for us tech guys ```UserPrincipalName``` does mean something. Moreover as a foreigner, I stumbled over the "user principal name" phrase for a second, wondering "What's the matter with the name of a user's manager?" =) The other purpose of having one word is you get better results on Google. Pretty much the same idea has led Jeffrey Snoover of Microsoft to the word ```cmdlet``` back in 2006, when he introduced PowerShell.
Nevertheless, I am really thankful for your efforts. You and many of your colleagues are making Microsoft documentation better every day!

**Comment by cabailey** [comment]
@YuriySamorodov  I didn't take it personally at all - always fun to learn together! Since the current text was signed off by the OneDrive PM, and it doesn't assume knowledge of Azure AD attribute names, we'll leave it like this for now and monitor for additional customer feedback. Thanks again for taking the time to flag this and improve the docs!

**Pull request closed without merging**